### PR TITLE
Merge23into30

### PR DESCRIPTION
**Pull request by jmleroux**

### CODE REVIEW

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountVariantProducts.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+namespace Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query;
+
+use Akeneo\Pim\Enrichment\Component\Product\ProductAndProductModel\Query\CountVariantProductsInterface;
+use Doctrine\DBAL\Connection;
+
+/**
+ * @copyright 2019 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ *
+ * @todo pull-up 3.x Move to `Akeneo\Pim\Enrichment\Bundle\Product\Query\Sql`
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountVariantProducts.php]: todo :)

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountProductModelsAndChildrenProductModels.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Akeneo\Pim\Enrichment\Bundle\Doctrine\ORM\Query;
```
Comment (mmetayer) [inline src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountProductModelsAndChildrenProductModels.php]: not sure, but probably the same todo as next comment

File: tests/back/Pim/Enrichment/Integration/Doctrine/Query/CountVariantProductsIntegration.php [hunk 3]
```diff
@@ -0,0 +1,244 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pim\Bundle\CatalogBundle\tests\integration\Doctrine\ORM\Query;
```
Comment (mmetayer) [inline tests/back/Pim/Enrichment/Integration/Doctrine/Query/CountVariantProductsIntegration.php]: wrong namespace (but this should be fixed in the base branch, 2.3)

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountVariantProducts.php (hunk 1 above)
Comment (jmleroux) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountVariantProducts.php]: It's for 3.1

File: src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountProductModelsAndChildrenProductModels.php (hunk 2 above)
Comment (jmleroux) [reply src/Akeneo/Pim/Enrichment/Bundle/Doctrine/ORM/Query/CountProductModelsAndChildrenProductModels.php]: For 3.1

**Pull request merged**